### ISSUE TITLE
Mojave support dropped in 1.4?

### ISSUE THREAD

**Issue by sirnacnud** [clementine-player/Clementine#6922]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: 10.14.6
- Clementine version: 1.4.0rc1-465-gb69dd2d90

### Expected behaviour / actual behaviour
I downloaded the lastest 1.4 release candidate and Catalina is required to run the app. Given that Apple is still supporting Mojave, would it be possible for 1.4 to support Mojave?

**Comment by jbroadus** [comment]
I don't have access to a Mojave system, but I took a shot at a build. Are you able to access this?
https://github.com/jbroadus/Clementine/suites/2038573028/artifacts/41000346

**Comment by sirnacnud** [comment]
I downloaded the zip, but the executable still requires Catalina.

I looked over the changes in the branch. I'm not familiar with this build system, but somehow the macOS deployment target needs to be lowered.

**Comment by jbroadus** [comment]
It fails to build with the CMake variable. But I'm giving this a try: https://github.community/t/selecting-an-xcode-version/16204

**Comment by jbroadus** [comment]
This looks more promising:
https://github.com/jbroadus/Clementine/suites/2040671090/artifacts/41045966

**Comment by sirnacnud** [comment]
I did a test build myself under my Catalina install with the below change at the start of `CMakeLists.txt` and the executable did work in my Mojave install. This way you have one executable that works for both. I will open a PR later.

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 600dc6efe..1a932618d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,7 @@
+if (APPLE)
+  set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14" CACHE STRING "Minimum OS X deployment target" FORCE)
+endif ()
+
 project(clementine)
 
 cmake_minimum_required(VERSION 2.8.11)
```

**Comment by hatstand** [comment]
Unfortunately you then need to make sure _every dependency_ is also built with that so it's not so simple.

**Comment by sirnacnud** [comment]
> Unfortunately you then need to make sure _every dependency_ is also built with that so it's not so simple.

Yeah I noticed some warnings when building since the dependencies were built with Catalina.

Since the dependencies are built with Homebrew, you can't specify the deployment target, and it will always use your current OS version.

**Comment by jbroadus** [comment]
As far as I can tell, setting `DEVELOPER_DIR` is doing the right thing. From the log:
```
 Found OpenGL: /Applications/Xcode_10.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework
```

**Comment by sirnacnud** [comment]
> This looks more promising:
> https://github.com/jbroadus/Clementine/suites/2040671090/artifacts/41045966

This build did work on Mojave!

**Comment by jbroadus** [comment]
Great. I'll open a pull request.

**Comment by jbroadus** [comment]
@sirnacnud Could you confirm that the builds are working for you now?

**Comment by sirnacnud** [comment]
@jbroadus The latest mac build worked on Mojave.Thanks for getting this fixed!
